data-processor/src/test/groovy/io/micronaut/data/processor/mappers/EntityAnnotationMapperSpec.groovy

```diff
@@ -74,6 +76,14 @@ class Test {
         introspection.hasStereotype(MappedEntity)
         introspection.getPropertyNames()
         introspection.getValue(MappedEntity, String).get() == 'test_tb1'
+        def tableAnnotation = introspection
+                .getAnnotationMetadata()
+                .findDeclaredAnnotation("javax.persistence.Table").get();
+        def indexes = (List<AnnotationValue<Index>>)tableAnnotation.getValues().get("indexes")
```

Comment:
Here you should work with a new annotation `@Indexes` and it should matter if the index came from `@Table`, entity annotation, or some other annotation.

Reply:
> Here you should work with a new annotation `@Indexes` and it should matter if the index came from `@Table`, entity annotation, or some other annotation.

Rather than doing that, I have preferred to allow a developer to use one single source for indexes in an entity.
That is the point of:
```
    private void addIndexes(PersistentEntity entity, String tableName, List<String> createStatements) {

        final String indexes = createIndexesStatementsFromTableAnnotation(entity, tableName);
        final String indexesFromFields = createIndexesStatementsFromFieldAnnotation(entity, tableName);
        final String createIndexFromClassAnnotations = createIndexFromClassAnnotations(entity, tableName);
        if (indexes.length() > 0) {
            createStatements.add(indexes);
        } else if (indexesFromFields.length() > 0) {
            createStatements.add(indexesFromFields);
        } else if (createIndexFromClassAnnotations.length() > 0) {
            createStatements.add(createIndexFromClassAnnotations);
        }
    }
```

Reply:
In the user-code which is the schema creator, it shouldn't matter where the annotations were added and for simplicity, it should just extract indexes from the entity type. The compile-time annotations processor should do most of the work. You would need to use `MappedEntityVisitor` to map field annotations to the entity type.

Reply:
OK, but then what happens if a user creates an index with a given name and in one case it is a composite index and in one case it is not?
Or what happens if a user adds indexes in the table annotation and indexes with the @index annotation, which is going to be the good one if I just keep appending?

Reply:
We should try to check invalid cases at the compile-time if it's possible